velox/vector/arrow/Bridge.cpp

```diff
@@ -576,14 +614,53 @@ void exportDictionary(
   }
   auto& values = *vec.valueVector()->loadedVector();
   out.dictionary = holder.allocateDictionary();
-  exportBase(values, Selection(values.size()), *out.dictionary, pool);
+  exportBase(
+      values, Selection(values.size()), *out.dictionary, pool, {false, false});
+}
+
+void exportFlattenedVector(
+    const BaseVector& vec,
+    const Selection& rows,
+    ArrowArray& out,
+    memory::MemoryPool* pool,
+    VeloxToArrowBridgeHolder& holder) {
+  VELOX_CHECK(vec.isScalar(), "Flattening is only supported for scalar types.");
+  VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
+      flattenAndExport, vec.typeKind(), vec, rows, out, pool, holder);
+}
+
+void exportDictionary(
```

Comment:
Separate functions for `exportDictionary` and `exportConstant` can be avoided. Let's inline the body inside `exportBase`.

Reply:
Yes, I had considered this but though it was better to have a wrapper. I'll change it.